### ISSUE TITLE
Does PyCBC actually need pillow?

### ISSUE THREAD

**Issue by duncanmmacleod** [gwastro/pycbc#3026]
PyCBC has a dependency on Pillow:

https://github.com/gwastro/pycbc/blob/60502a8a097e0692c9930cc9f3f0816e46eb6c4b/setup.py#L45

but I can't see that being used anywhere. Can this dependency be removed?

**Comment by ahnitz** [comment]
Yes, we use this when generating result plots. It allows us to store all our metadata directly in the images (such as captions, titles, and links, etc).

**Comment by duncanmmacleod** [comment]
So this is an optional dependency of ... matplotlib?

**Comment by ahnitz** [comment]
At the moment, it has the same level of requirement as matplotlib. It would be straightforward though to block that off to be even more optional (i.e so not even plotting codes break without it).

**Comment by ahnitz** [comment]
At that point, I'd be comfortable removing it as an library dependence and just having it in the requirements.txt file. Is that what you are after?

**Comment by duncanmmacleod** [comment]
To be honest, I was just curious, its in the build, but I didn't see any references. It's not causing any problems, so maybe just a note in the `setup.py` to explain why it is there would ward off more people like me.

**Comment by ahnitz** [comment]
@duncan-brown Eh, easy enough to reduce the dependency to optional. PR here #3027
